### DISCUSSION TITLE
Non linear programming

### DISCUSSION THREAD

**Discussion by khadjiyi** [optimatika/ojAlgo#481] [General, ojAlgo and related code]
I am interested in non linear  programming. As far as I can understand this is not directly provided from ojAlgo. 
Any plans to provide a sequential quadratic programming for example?

**Comment by apete** [comment]
Thinking about it, yes.

I've long thought that it's a bit of a problem that the current set of built-in solvers can't handle anything/everything ExpressionsBasedModel can model (can't handle quadratic constraints). That, at least, would be very nice to have.

Are you willing to contribute work towards such solver?

**Comment by khadjiyi** [comment]
I need to investigate it first and let you know. SQP is a priority. Primal dual interior point method will also be nice to have, JOptimizer at least provides it.

**Comment by khadjiyi** [comment]
I was wondering if we have any update on that issue.

**Reply by apete** [reply]
No, not really. The solvers have been improved, but not to handle new problem types.
